storage/src/main/java/fk/prof/storage/AsyncStorage.java

```diff
@@ -0,0 +1,27 @@
+package fk.prof.storage;
+
+import java.io.InputStream;
+import java.util.concurrent.Future;
+
+/**
+ * Interface for a simple async load/store storage.
+ * @author gaurav.ashok
+ */
+public interface AsyncStorage {
+
+    /**
+     * Store the content to the specified path. IOException while storing
+     * needs to be taken care of by the implementation.
+     * @param path
+     * @param content
+     */
+    void store(String path, InputStream content);
```

Comment:
Looks like bucket can be a param here rather than asyncstorage instance being tied to a d42 bucket. In case we have multiple d42 objects, the same instance can work for all. downside of that is that asyncstorage interface becomes specific to bucket-esque storages (like d42).
Your call